app/controllers/webhooks_controller.rb

```diff
@@ -0,0 +1,65 @@
+class WebhooksController < ApplicationController
+  skip_before_action :verify_authenticity_token, only: [ :stripe ]
+  skip_authentication
+
+  def stripe
+    webhook_body = request.body.read
+    sig_header = request.env["HTTP_STRIPE_SIGNATURE"]
+    client = Stripe::StripeClient.new(ENV["STRIPE_SECRET_KEY"])
+
+    begin
+      thin_event = client.parse_thin_event(webhook_body, sig_header, ENV["STRIPE_WEBHOOK_SECRET"])
+
+      event = client.v1.events.retrieve(thin_event.id)
+
+      case event.type
+      when /^customer\.subscription\./
+        handle_subscription_event(event)
+      when "customer.created", "customer.updated", "customer.deleted"
+        handle_customer_event(event)
+      else
+        Rails.logger.info "Unhandled event type: #{event.type}"
+      end
+
+    rescue JSON::ParserError
+      render json: { error: "Invalid payload" }, status: :bad_request
+      return
+    rescue Stripe::SignatureVerificationError
+      render json: { error: "Invalid signature" }, status: :bad_request
+      return
+    rescue => e
+      Rails.logger.error "Error processing Stripe webhook: #{e.message}"
+      render json: { error: "Internal server error" }, status: :internal_server_error
+      return
```

Comment:
We'll probably want to throw all this handling in a background job at some point, so I'd probably remove this "catch all" handler and let Sentry capture anything that reaches this point.  That way this endpoint remains solely responsible for parsing the payload and verifying the signature.